server/apps/immich/src/api-v1/asset/asset-repository.ts

```diff
@@ -238,6 +239,30 @@ export class AssetRepository implements IAssetRepository {
     });
   }
 
+  /**
+   * Get all assets belong to the user on the database and that contain exif location information
+   * @param ownerId
+   */
+  async getMapMarkerByUserId(ownerId: string): Promise<AssetEntity[]> {
+    return this.assetRepository.find({
+      where: {
```

Comment:
This query selects all fields of the asset and exif entities by default, which takes more time and can lead to high memory consumption. I would recommend to only select the fields that are actually used:

```suggestion
    return this.assetRepository.find({
      select: {
        id: true,
        exifInfo: {
          latitude: true,
          longitude: true,
        },
      },
      where: {
```